### PR TITLE
F/copy with flag

### PR DESCRIPTION
**Pull request by mellonnen**
-  cleanup in main package
- replace -p and -a flags with --relay flag 
- add behavior discussed in #46

### CODE REVIEW

File: cmd/portal/main.go [hunk 1]
```diff
@@ -20,6 +20,21 @@ import (
 // injected at link time using -ldflags.
 var version string
 
+// Initialization of cobra and viper.
+func init() {
+	cobra.OnInitialize(initViperConfig)
+
+	rootCmd.PersistentFlags().BoolP("verbose", "v", false, "Specifes if portal logs debug information to a file on the format `.portal-[command].log` in the current directory")
```
Comment (ZinoKader) [inline cmd/portal/main.go:27]: Specifes -> Specifies

Maybe shorten this to "Log debug information to a file with the format `.portal-[command].log` in the current directory"

File: cmd/portal/send.go [hunk 2]
```diff
@@ -12,18 +12,27 @@ import (
 	"github.com/spf13/viper"
 )
 
+// Set flags.
+func init() {
+	// Add subcommand flags (dummy default values as default values are handled through viper)
+	sendCmd.Flags().StringP("relay", "r", "", "address of the relay server")
```
Comment (ZinoKader) [inline cmd/portal/send.go:18]: I think we should document the format which the relay flag expects more clearly. Maybe something like "address of the relay server (host:port)"?

File: cmd/portal/send.go [hunk 3]
```diff
@@ -40,30 +49,26 @@ var sendCmd = &cobra.Command{
...
 	// Conditionally add option to sender ui
 	if err == nil {
 		opts = append(opts, senderui.WithVersion(ver))
 	}
-	sender := senderui.New(fileNames, fmt.Sprintf("%s:%d", addr, port), opts...)
+	relayAddr := viper.GetString("relay")
+	if selfHostedRelay {
+		opts = append(opts, senderui.WithCopyFlags(map[string]string{"--relay": relayAddr}))
+	}
+	sender := senderui.New(fileNames, relayAddr, opts...)
```
Comment (ZinoKader) [inline cmd/portal/send.go:71]: I'm curious about an alternative approach for this part. What do you think about gettig the `relayAddr` using `viper.getString("relay")` inside the `copyReceiverCommand` method in senderui?

We could then (if I understand the usage of this correctly) see if has changed from the default using https://pkg.go.dev/github.com/spf13/viper#FlagValue.HasChanged

What do you think?

**Comment by ZinoKader** [comment]
I don't think `validateRendezvousAddressInViper` is working anymore as it's using `viper.GetString("rendezvousAddress")`

File: cmd/portal/send.go (hunk 3 above)
Comment (mellonnen) [reply cmd/portal/send.go:71]: Effectively, we are already doing this right? Check 
 
https://github.com/SpatiumPortae/portal/blob/8deae65d730f515b2fe3a94499d24b152fb65d2c/cmd/portal/send.go#L52

`GetString` only returns the string value associated with the key, not a `FlagValue` so we cannot check if it has changed. From what I can gather we have to get the flag values from the cobra command in order to check if they have changed.

Also, this approach allows us to add more flags in the future if we want to.

File: cmd/portal/send.go (hunk 2 above)
Comment (mellonnen) [reply cmd/portal/send.go:18]: It does not necessarily have to be (host:port) right? How would you explain that? 🤔

File: cmd/portal/send.go (hunk 3 above)
Comment (ZinoKader) [reply cmd/portal/send.go:71]: Yes I saw that we already are doing that. I just hoped there'd be a way to not pass arounf:so much information if it can be gotten where it is needed.

File: cmd/portal/send.go (hunk 2 above)
Comment (mellonnen) [reply cmd/portal/send.go]: [2883d8d](https://github.com/SpatiumPortae/portal/pull/47/commits/2883d8d918bca049cc14c038a28e416a3cb88d27)

File: cmd/portal/send.go (hunk 3 above)
Comment (mellonnen) [reply cmd/portal/send.go:77]: Hmm I see, we could perhaps directly access it and check if the relay address differs from the default one, But I do not really know how that would interact with user configs. IMO I think it is fine passing it as an option, what do you think?

**Comment by mellonnen** [comment]
> I don't think validateRendezvousAddressInViper is working anymore as it's using viper.GetString("rendezvousAddress")

[a095d42](https://github.com/SpatiumPortae/portal/pull/47/commits/a095d4269f6c944fb5aaed0e4686e81c5ff303ba)

File: cmd/portal/send.go (hunk 3 above)
Comment (ZinoKader) [reply cmd/portal/send.go:77]: I think it's fine too -- although it might be worth it to just check how this interacts with the changes in #50. I have a feeling that changing the `relay` in the config file to something other than the default would not give you the copy command with the appropriate flags.

**Review by ZinoKader — APPROVED** [review]
